corehq/apps/app_manager/xform_builder.py

```diff
@@ -0,0 +1,432 @@
+... '''          <text id="personal/fav_color-label">\\n'''
+... '''            <value>'''
+...               '''Quelle est ta couleur pr\xc3\x83\xc2\xa9f\xc3\x83\xc2\xa9r\xc3\x83\xc2\xa9e?'''
+...             '''</value>\\n'''
+... '''          </text>\\n'''
+... '''          <text id="personal/fav_color-r-label">\\n'''
+... '''            <value>Rot</value>\\n'''
+... '''          </text>\\n'''
+... '''          <text id="personal/fav_color-b-label">\\n'''
+... '''            <value>Blau</value>\\n'''
+... '''          </text>\\n'''
+... '''          <text id="personal/fav_color-g-label">\\n'''
+... '''            <value>Gr\xc3\x83\xc2\xbcn</value>\\n'''
+... '''          </text>\\n'''
+... '''        </translation>\\n'''
+... '''      </itext>\\n'''
+... '''      <bind nodeset="/data/name" type="xsd:string"/>\\n'''
+... '''      <bind nodeset="/data/personal"/>\\n'''
+... '''      <bind nodeset="/data/personal/fav_color" type="xsd:string"/>\\n'''
+... '''    </model>\\n'''
+... '''  </h:head>\\n'''
+... '''  <h:body>\\n'''
+... '''    <input ref="/data/name">\\n'''
+... '''      <label ref="jr:itext(\'name-label\')"/>\\n'''
+... '''    </input>\\n'''
+... '''    <group ref="/data/personal">\\n'''
+... '''      <label ref="jr:itext(\'personal-label\')"/>\\n'''
+... '''      <input ref="/data/personal/fav_color">\\n'''
+... '''        <label ref="jr:itext(\'personal/fav_color-label\')"/>\\n'''
+... '''      </input>\\n'''
+... '''    </group>\\n'''
+... '''  </h:body>\\n'''
+... '''</h:html>\\n''')
+True
+
+"""
+import uuid
+from lxml import etree
+from lxml.builder import E
+
+EMPTY_XFORM = """<?xml version="1.0"?>
+<h:html xmlns:h="http://www.w3.org/1999/xhtml"
+        xmlns:orx="http://openrosa.org/jr/xforms"
+        xmlns="http://www.w3.org/2002/xforms"
+        xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+        xmlns:jr="http://openrosa.org/javarosa">
+    <h:head>
+        <h:title>{name}</h:title>
+        <model>
+            <instance>
+                <data xmlns:jrm="http://dev.commcarehq.org/jr/xforms"
+                      xmlns="{xmlns}"
+                      uiVersion="1"
+                      version="3"
+                      name="{name}"/>
+            </instance>
+            <itext>
+                <translation lang="en" default=""/>
+            </itext>
+        </model>
+    </h:head>
+    <h:body/>
+</h:html>"""
+
+
+# The data types supported by the Open Data Kit XForm spec
+# cf. https://opendatakit.github.io/odk-xform-spec/#data-types
+ODK_TYPES = (
+    'string', 'int', 'boolean', 'decimal', 'date', 'time', 'dateTime', 'select', 'select1', 'geopoint', 'geotrace',
+    'geoshape', 'binary', 'barcode',
+)
+# CommCare question group types
+GROUP_TYPES = ('group', 'repeatGroup')  # TODO: Support 'questionList'
+# The subset of ODK data types that are XSD data types
+# cf. http://www.w3.org/TR/xmlschema-2/#built-in-datatypes
+XSD_TYPES = ('string', 'int', 'boolean', 'decimal', 'date', 'time', 'dateTime')
```

Comment:
Nitpick: you could eliminate a tiny bit of duplication by defining this before `ODK_TYPES` and then
```py
ODK_TYPES = XSD_TYPES + ('select', 'select1', ...
```